### PR TITLE
UI improvements; read and delete operations for schedules

### PR DESCRIPTION
**Pull request by wilsonwid**
This PR adds various UI improvements, including the ability to display schedules. This also adds read and delete operations for schedules, which fulfils the basic CRUD functionalities required for schedules.

This PR also allows for saving to the hard disk of schedules.

### CODE REVIEW

File: src/main/java/scm/address/logic/Logic.java [hunk 1]
```diff
@@ -47,4 +48,7 @@ public interface Logic {
      * Set the user prefs' GUI settings.
      */
     void setGuiSettings(GuiSettings guiSettings);
+
+    /** Returns an unmodifiable view of the list of schedules. **/
```
Comment (ChuaZiLong) [inline src/main/java/scm/address/logic/Logic.java:52]: I like that the Javadocs explains a key characteristic of the list

File: src/main/java/scm/address/logic/commands/DeleteScheduleCommand.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    private final Index targetIndex;
+
+    /**
+     * Constructs a DeleteScheduleCommand.
+     *
+     * @param targetIndex The target index to delete.
+     */
+    public DeleteScheduleCommand(Index targetIndex) {
+        requireNonNull(targetIndex);
+        this.targetIndex = targetIndex;
```
Comment (ChuaZiLong) [inline src/main/java/scm/address/logic/commands/DeleteScheduleCommand.java:36]: I like that the variable name is descriptive

File: src/main/java/scm/address/logic/commands/DeleteScheduleCommand.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        requireNonNull(model);
+        List<Schedule> lastShownList = model.getFilteredScheduleList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_SCHEDULE_DISPLAYED_INDEX);
+        }
+
+        Schedule scheduleToDelete = lastShownList.get(targetIndex.getZeroBased());
+        model.removeSchedule(scheduleToDelete);
+        return new CommandResult(String.format(MESSAGE_DELETE_SCHEDULE_SUCCESS, Messages.format(scheduleToDelete)));
```
Comment (ChuaZiLong) [inline src/main/java/scm/address/logic/commands/DeleteScheduleCommand.java:50]: I like that constants are being used so it is more readable

File: src/main/java/scm/address/logic/parser/AddressBookParser.java [hunk 4]
```diff
@@ -93,6 +95,12 @@ public Command parseCommand(String userInput) throws ParseException {
         case EditScheduleCommand.COMMAND_WORD:
             return new EditScheduleCommandParser().parse(arguments);
 
+        case ListScheduleCommand.COMMAND_WORD:
```
Comment (ChuaZiLong) [inline src/main/java/scm/address/logic/parser/AddressBookParser.java:98]: Good use of switch cases for control flow

File: src/test/java/scm/address/logic/commands/AddScheduleCommandTest.java [hunk 5]
```diff
@@ -149,7 +149,11 @@ public void toStringAccurate() {
                 startDateTime,
                 endDateTime);
         AddScheduleCommand command1 = new AddScheduleCommand(schedule1);
-
-        assertTrue(command1.toString().equals("MeetingProject Discussion2023-03-21T15:002023-03-21T16:00"));
+        String expectedOutput = Schedule.class.getCanonicalName()
```
Comment (ChuaZiLong) [inline src/test/java/scm/address/logic/commands/AddScheduleCommandTest.java:152]: Thanks for fixing my test for me

**Review by ChuaZiLong — APPROVED** [review]

File: src/test/java/scm/address/logic/commands/AddScheduleCommandTest.java (hunk 5 above)
Comment (wilsonwid) [reply src/test/java/scm/address/logic/commands/AddScheduleCommandTest.java:152]: No worries!

**Pull request merged**